lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/viewmodel/FindMealViewModel.kt

```diff
@@ -1,7 +1,86 @@
 package com.kinandcarta.lib.find.meal.viewmodel
 
+import android.location.Location
+import androidx.annotation.RequiresPermission
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.kcc.kmmhackathon.shared.MealsSDK
+import com.kcc.kmmhackathon.shared.entity.Meal
+import com.kcc.kmmhackathon.shared.utility.DistanceUnit
+import kotlinx.coroutines.launch
 
-class FindMealViewModel : ViewModel() {
-    // TODO: Implement the ViewModel
+typealias Meals = List<Meal>
+
+class FindMealViewModel @ViewModelInject constructor(
+    private val fusedLocationClient: FusedLocationProviderClient
+) : ViewModel() {
+
+    sealed class State {
+        object LoadingMeals : State()
```

Comment:
We could add a "refreshing meals" state to this if we needed to differentiate in the UI between initial load and pull to refresh etc

Reply:
Use of State here is really cool 🤩

Reply:
That's a good shout! It's easy to add an empty state now, just need to add a new object and then handle it in the UI